### PR TITLE
Fixed incorrect output path concatenation when cusom output path is provided

### PR DESCRIPTION
**Pull request by Juriyx**
Command like `dotnet build -o c:\Temp\wasm\` produced error `error : System.IO.IOException: The filename, directory name, or volume label syntax is incorrect. : '\\?\c:\Temp\wasm\/dist\package_50d3e66f6ede02ff1eb699943f97d000fa3256fd'`

Fixed

### CODE REVIEW

**Comment by Juriyx** [comment]
@jeromelaban, how should we cover this case with tests?

**Comment by Juriyx** [comment]
Could you issue a patched NuGet package with this fix, please?

File: src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.Publish.targets [hunk 1]
```diff
@@ -13,7 +13,7 @@
         <TargetPath>$([System.String]::Copy(%(TargetPath)).Substring(8))</TargetPath>
       </ContentWithTargetPath>
 
-      <_UnoWasmDistFiles Include="$(OutputPath)dist\**"/>
+      <_UnoWasmDistFiles Include="$(OutputPath)dist/**"/>
```
Comment (jeromelaban) [inline src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.Publish.targets:16]: This likely will break the publication on windows. We'll need to Add a test that validates the output on both platforms.

**Comment by jeromelaban** [comment]
> @jeromelaban, how should we cover this case with tests?

For tests, I generally either make an app that changes its output path explicitly and validates it using `AfterTargets="Build"`, or make a specific step in the ci scripts to validate that the output is generated at a custom location.

File: src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.Publish.targets (hunk 1 above)
Comment (xperiandri) [reply src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.Publish.targets:16]: No, it works fine. I tested

Comment (xperiandri) [reply src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.Publish.targets:16]: Yes, tests are necessary

**Comment by xperiandri** [comment]
Do you want us to do that (requires instructions) or will the Uno team handle that?

**Comment by xperiandri** [comment]
🎉 pipelines succeeded

File: src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.Publish.targets (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.Wasm.Bootstrap/build/Uno.Wasm.Bootstrap.Publish.targets:16]: This change is actually ok, it's an item group, so msbuild handles both slash directions.

**Comment by jeromelaban** [comment]
So I've been looking at this, and the Windows support for AOT got this fixed implicitly. So I'll merge this as-is but will backport it to 2.1.

Thanks !

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/2.1

**Pull request merged**